gateway/api_definition.go

```diff
@@ -1084,84 +1084,84 @@ func (a *APISpec) CheckSpecMatchesStatus(r *http.Request, rxPaths []URLSpec, mod
 	}
 
 	// Check if ignored
-	for _, v := range rxPaths {
-		if mode != v.Status {
+	for i := range rxPaths {
+		if mode != rxPaths[i].Status {
```

Comment:
the same idea - would be good to see benchmark difference vs master where this code involved, i.e. `BenchmarkResponseHeaderInjection`